src/directives/formly-form.js

```diff
@@ -226,13 +239,23 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
       angular.forEach($scope.fields, (field) => {
         const isNewModel = initModel(field)
 
-        if (field.model && isNewModel && watchedModels.indexOf(field.model) === -1) {
+        if (field.model && isNewModel && watchedModels.indexOf(field.model) === -1 && !manualModelWatcher) {
           $scope.$watch(() => field.model, onModelOrFormStateChange, true)
           watchedModels.push(field.model)
         }
       })
     }
 
+    function setupExpressionWatchers(field, index) {
```

Comment:
This only sets up a watcher for `hideExpression`. Perhaps the function should be called `setupHideExpressionWatcher`